### PR TITLE
fix: add validation on data.method when using transport.request

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/services/CommonService.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ import {
 } from "../../../../src/core/server";
 import { IAPICaller } from "../../models/interfaces";
 
+const VALID_METHODS = ["HEAD", "GET", "POST", "PUT", "DELETE"];
+
```
Comment (Hailong-am) [inline server/services/CommonService.ts:18]: should we support all standard http methods mentioned here https://www.rfc-editor.org/rfc/rfc7231#section-4.1. ?

CONNECT   
DELETE   
GET  
HEAD  
OPTIONS 
POST  
PUT 
TRACE

Comment (Hailong-am) [reply server/services/CommonService.ts:18]: sounds good.

**Review by Hailong-am — APPROVED** [review]

**Comment by xluo-aws** [comment]
@SuZhou-Joe , could you add a UT for it?

File: server/services/CommonService.ts [hunk 2]
```diff
@@ -36,12 +38,42 @@ export default class IndexService {
...
        */
       if (endpoint === "transport.request" && typeof finalData?.path === "string" && !/^\//.test(finalData?.path || "")) {
         finalData.path = `/${finalData.path || ""}`;
       }
+
+      /**
+       * Check valid method here
+       */
+      if (endpoint === "transport.request" && data?.method) {
+        if (VALID_METHODS.indexOf(data.method?.toUpperCase()) === -1) {
```
Comment (xluo-aws) [inline server/services/CommonService.ts:66]: ? after data.method can be removed because it's checked for null in 65

**Comment by SuZhou-Joe** [comment]
> @SuZhou-Joe , could you add a UT for it?

Sure, thanks for reminding.

File: server/services/CommonService.ts (hunk 2 above)
Comment (SuZhou-Joe) [reply server/services/CommonService.ts:66]: Yes, good point and modify that to catch toUpperCase in case data.method is not a string.

**Review by xluo-aws — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**